src/halmos/__main__.py

```diff
@@ -820,37 +829,44 @@ def package_result(
             print(f"  Valid path; counterexample generated (path id: {idx+1})")
 
         # convert model into string to avoid pickling errors for z3 (ctypes) objects containing pointers
-        validity = None
+        is_valid = None
         if model:
             if isinstance(model, str):
-                validity = True
+                is_valid = True
                 model = f"see {model}"
             else:
-                validity = is_valid_model(model)
-                model = f"{str_model(model, args)}"
+                is_valid = is_model_valid(model)
+                model = to_str_model(model, args.print_full_model)
 
-        return ModelWithContext(model, validity, idx, result)
+        return ModelWithContext(model, is_valid, idx, result)
 
     else:
         if args.verbose >= 1:
             print(f"  Timeout (path id: {idx+1})")
         return ModelWithContext(None, None, idx, result)
 
 
-def is_valid_model(model) -> bool:
+def is_model_valid(model: AnyModel) -> bool:
     for decl in model:
         if str(decl).startswith("evm_"):
             return False
     return True
 
 
-def str_model(model, args: Namespace) -> str:
+def to_str_model(model: Model, print_full_model: bool) -> StrModel:
     def select(var):
         name = str(var)
         return name.startswith("p_") or name.startswith("halmos_")
 
-    select_model = filter(select, model) if not args.print_full_model else model
-    formatted = [f"\n    {decl} = {hexify(model[decl])}" for decl in select_model]
+    select_model = filter(select, model) if not print_full_model else model
+    return {str(decl): hexify(model[decl]) for decl in select_model}
```

Comment:
hexify is not needed here because it will be done in render_model(), right?

Reply:
let's keep it in here so that we have hexified output in the json